v2/internal/genericarmclient/error_classification.go

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package genericarmclient
+
+type ErrorClassification string
```

Comment:
minor: So originally this was in `reconcilers`, which probably caused issues with circular references?

I'm cool with _this_ being in `core` if you want, as _this_ is the reconcilers classification of an error as retryable or fatal. (It's not as about the wire format as the other stuff).

I'm also OK with it here though, but just wanted to clarify that if you felt this made more sense in `core` I could get behind that, since it's not wire-format stuff but rather instructions for the reconciler(s).

Reply:
> which probably caused issues with circular references?

Yes! I'd forgotten the details. Will move this back to `core`.